dashboard/views.py

```diff
@@ -134,6 +137,90 @@ def get_course_info(request, course_id=0):
     return HttpResponse(json.dumps(resp, default=str))
 
 
+@permission_required('dashboard.update_course_info',
+                     fn=objectgetter(Course, 'course_id', 'canvas_id'), raise_exception=True)
+def update_course_info(request, course_id=0):
+    """
+
+    :param request: HTTP `PUT` req.; body should contain the JSON body…
+    :param course_id: Integer Canvas course ID number, typically six digits or less.
+    :return: HttpResponse containing `{"default": "success"}` or `{"default": "fail"}`
+    """
+    logger.info(update_course_info.__name__)
+
+    if (request.method != 'PUT'):
+        return HttpResponseBadRequest(JsonResponse({'error': 'Invalid request method.'}))
+
+    course_id = canvas_id_to_incremented_id(course_id)
+    current_user = request.user.get_username()
+
+    bad_json_response = HttpResponseBadRequest(JsonResponse({'error': 'Request JSON malformed.'}))
```

Comment:
Something about this constructor chaining is leading to the `Content-Type` header being changed to `text/html; charset=utf-8` from `application/json`. It's not an issue with the successful response returned, so the `HttpResponseBadRequest` must be converting it. Maybe instead of using this class, you can just set the `status` in the constructor, i.e. `JsonResponse({'error': 'Request JSON malformed.'}, status=400)`.

Not a huge deal, but I thought I'd point it out. You can see the header with `response.__getitem__('Content-Type')`. If you do fix this, please remove the `HttpResponseBadRequest` import.

Reply:
OK.  That's my bad.  I'd looked for a way to set the status code with `JsonResponse`, but I didn't find it.  I must have overlooked it in the documentation and examples.

Reply:
I think its available because it's a subclass of `HttpResponse`.

Reply:
The IDE import reorganization feature that I used again to address your `math` comment removed `HttpResponseBadRequest` for me.